<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Functional interpretation of single cell RNA-seq latent manifolds • VISION</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Functional interpretation of single cell RNA-seq latent manifolds">
<meta property="og:description" content="VISION provides functional interpretation of single cell RNA-seq (scRNA-seq) latent manifolds through the use of biological signatures (which can be downloaded from online) and precomputed metadata such as clustering assignments, disease status of a cell, or the number of UMIs observed in a cell. VISION can operate downstream of other common analyses such as dimensionality reduction, clustering, or trajectory analysis of scRNA-seq data. VISION produces an interactive web-based output report that can be easily shared with other collaborators or the greater scientific community with ease.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">VISION</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.99.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/VISION-vignette.html">Introduction to VISION</a>
    </li>
    <li>
      <a href="articles/micropooling.html">Micropooling and VISION</a>
    </li>
    <li>
      <a href="articles/trajectories.html">Trajectory Inference and VISION</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="contents col-md-9">
    <div id="vision-travis-ci-build-status" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#vision-travis-ci-build-status" class="anchor"></a>VISION <a href="https://travis-ci.org/YosefLab/VISION"><img src="https://travis-ci.org/YosefLab/VISION.svg?branch=master" alt="Travis-CI Build Status"></a>
</h1></div>
<p>Here we present VISION, a module that can sit downstream of other common analyses such as clustering, dimensionality reduction, and trajectory inference. Specifically, VISION aids in the interpretation of scRNA-seq data, with or without predetermined labels, or stratifications of the data (e.g. clusterings) using the notion of cell-cell similarity maps (as interpreted from some latent space) and biological signatures (functional sets of genes that can be obtained online from, for example, MSigDB). Finally, VISION can evaluate the effect of cell- level meta data, such as library quality, batch, clinical information, or additional experimental readouts (e.g., protein levels from a Cite-Seq experiment). Importantly, the use of VISION can greatly facilitate collaborative projects, as it offers a low- latency interactive report for the end- user, which can be hosted online and viewed on a web browser without the need for installing developer-grade software.</p>
<p><em>For more information, refer to our preprint on biorxiv: <a href="https://www.biorxiv.org/content/early/2018/08/29/403055">Functional Interpretation of Single-Cell Similarity Maps</a></em></p>
<div id="installing-vision" class="section level2">
<h2 class="hasAnchor">
<a href="#installing-vision" class="anchor"></a>Installing VISION</h2>
<p>We recommend installing VISION via github using devtools:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">require</span>(devtools)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="st">"YosefLab/VISION"</span>, <span class="dt">build_vignettes=</span>T)</a></code></pre></div>
<p>We recommend using the build_vignettes argument to view vignettes from R (use the <code>vignette()</code> function).</p>
</div>
<div id="the-vision-pipeline" class="section level2">
<h2 class="hasAnchor">
<a href="#the-vision-pipeline" class="anchor"></a>The VISION Pipeline</h2>
<p>VISION generally follows the same pipeline from iteration to iteration, where minor differences can be specified via the various parameters in a VISION object. On a typical VISION run:</p>
<ul>
<li>For large datasets, or if the user so chooses, micropools are computed - grouping similar cells together to reduce the complexity of the analysis.</li>
<li>If a latent space is not specified, PCA is performed and the top 30 components are retained.</li>
<li>A KNN graph is constructed from the latent space, named the cell-cell similarity map</li>
<li>Signature scores are computed using the expression matrix</li>
<li>Signature local “consistencies” on the cell-cell similarity map are computed using the Geary-C statistic, an auto-correlation statistic.</li>
<li>An interactive web-based report is generated that can be used to explore and interpret the dataset.</li>
</ul>
</div>
<div id="how-to-run-vision" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-run-vision" class="anchor"></a>How to run VISION</h2>
<p>You can refer to the <a href="/vignettes">vignettes</a> to run VISION. To note, there is an extra vignette detailing how to properly interface with <a href="https://github.com/dynverse">Dynverse</a> for incorporating VISION into your trajectory inference pipeline.</p>
</div>
<div id="sample-output" class="section level2">
<h2 class="hasAnchor">
<a href="#sample-output" class="anchor"></a>Sample Output</h2>
<p><a href="http://s124.millennium.berkeley.edu:7703/">Click here for an example output report of ~9,000 CBMC’s sequenced with the CITE-seq protocol</a> <img src="https://github.com/YosefLab/VISION/blob/master/docs/example_report.png" alt="Screenshot of report"></p>
</div>
</div>
  </div>

  <div class="col-md-3" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Report a bug at <br><a href="https://github.com/YosefLab/VISION/issues">https://​github.com/​YosefLab/​VISION/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Matt Jones <br><small class="roles"> Author </small>  </li>
<li>Tal Ashuach <br><small class="roles"> Author, maintainer </small>  </li>
<li>David Detomaso <br><small class="roles"> Author </small>  </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Matt Jones, Tal Ashuach, David Detomaso.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
