<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to PhyloVision and Hotspot • VISION</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Introduction to PhyloVision and Hotspot">
<meta property="og:description" content="VISION">
<meta property="og:image" content="https://yoseflab.github.io/VISION/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">VISION</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/VISION-vignette.html">Getting Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Signatures.html">Gene Signatures</a>
    </li>
    <li>
      <a href="../articles/web_only/SignatureAutocorrelation.html">Signature Autocorrelation</a>
    </li>
    <li>
      <a href="../articles/micropooling.html">Micropooling</a>
    </li>
    <li>
      <a href="../articles/trajectories.html">Incorporating a Trajectory Model</a>
    </li>
    <li>
      <a href="../articles/phyloVision.html">PhyloVision and Hotspot</a>
    </li>
    <li>
      <a href="../articles/spatialHotspot.html">Spatial Data and Hotspot</a>
    </li>
    <li>
      <a href="../articles/web_only/Seurat.html">How To: Working with Seurat</a>
    </li>
    <li>
      <a href="../articles/web_only/10xData.html">How To: 10x CellRanger Outputs</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/YosefLab/VISION/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to PhyloVision and Hotspot</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/YosefLab/VISION/blob/HEAD/vignettes/phyloVision.Rmd" class="external-link"><code>vignettes/phyloVision.Rmd</code></a></small>
      <div class="hidden name"><code>phyloVision.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>As of version 3.0.0, VISION now supports the analysis of scRNA-seq
data with respect to a user-defined cell lineage, or phylogeny (e.g.,
the CRISPR/Cas9-based lineage tracing technologies described in <a href="https://www.nature.com/articles/s41586-019-1184-5" class="external-link">Chan et al,
Nature 2019</a>). We refer to this analysis pipeline as PhyloVision.
PhyloVision uses the relationships between cells as specified by the
phylogeny, as opposed to the cell-cell similarities that can be inferred
from the scRNA-seq data, to conduct the autocorrelation analysis. In
this way, the analyses performed in PhyloVision identify gene modules
that are heritable, or structured meaningfully on a cell lineage.</p>
<p>While several types of trees can be used to relate cells to one
another (e.g., those inferred via B Cell Receptor sequencing), in this
vignette, we focus on applying PhyloVision to a recently published
dataset using CRISPR/Cas9-based synthetic lineage tracing from <a href="https://www.science.org/doi/abs/10.1126/science.abc1944" class="external-link">Quinn et
al, Science 2021</a>. This technology relies on Cas9 to introduce
heritable, irreversible mutations at a defined locus in the 3’ UTR of a
fluorescent protein (“target site”) across cell divisions, encoding
nested relationships between cells. Utilizing conventional scRNA-seq
platforms, this data simultaneously captures single-cell transcriptomes
as well as target sites harboring lineage barcodes. Using this dataset,
we demonstrate how a user might use PhyloVision.</p>
<p>Specifically, we discuss four items: - Preparing data for input into
<code>PhyloVision</code>, including passing external meta-data (e.g.,
metastatic rates) to the pipeline. - Performing the
<code>PhyloVision</code> analysis pipeline - Detecting de-novo gene
modules with Hotspot (discussed more in the <a href="spatialHotspot.html">Spatial Hotspot Vignette</a>). - Launching an
interactive, web-based report for viewing.</p>
</div>
<div class="section level2">
<h2 id="preliminaries">Preliminaries<a class="anchor" aria-label="anchor" href="#preliminaries"></a>
</h2>
<p>If you have yet to install VISION, we recommend installing the
package from Github to install this package. Full source code can be
found at the VISION Github repository, available <a href="http://www.github.com/YosefLab/VISION" class="external-link">here</a>.</p>
<pre class="r{eval=false}"><code><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span>
<span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"YosefLab/VISION"</span><span class="op">)</span></code></pre>
<p>You’ll also want to be sure that you have <code>ape</code>,
<code>Matrix</code>, and <code>reticulate</code> installed:</p>
<pre class="r{eval=false}"><code><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'ape'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'Matrix'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'reticulate'</span><span class="op">)</span></code></pre>
<p>Once VISION and R are installed, you may load in VISION using
<code><a href="https://yoseflab.github.io/VISION">library(VISION)</a></code>.</p>
<p>To enable the Hotspot analysis below, install it directly from the
git repository using the following command:</p>
<pre><code>pip install git+https://github.com/yoseflab/Hotspot.git</code></pre>
<p>If you are having trouble installing Hotspot, please refer to the
documentation website <a href="https://yoseflab.github.io/Hotspot/" class="external-link">here</a>.</p>
</div>
<div class="section level2">
<h2 id="running-phylovision">Running PhyloVision<a class="anchor" aria-label="anchor" href="#running-phylovision"></a>
</h2>
<p>First, we need to load VISION, reticulate and ape.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>echo <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yoseflab.github.io/VISION">VISION</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/reticulate" class="external-link">reticulate</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://ape-package.ird.fr/" class="external-link">ape</a></span><span class="op">)</span></code></pre></div>
<div class="section level3">
<h3 id="creating-a-phylovision-object-with-a-tree">Creating a PhyloVision Object with a Tree<a class="anchor" aria-label="anchor" href="#creating-a-phylovision-object-with-a-tree"></a>
</h3>
<p>We’ll proceed by reading in the raw expression matrix, tree, and meta
data for a single clone (CP007) of approximately 600 cells from <a href="https://www.science.org/doi/abs/10.1126/science.abc1944" class="external-link">Quinn et
al, Science 2021</a>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Read in expression data</span>
<span class="va">expression</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">'data/metastasis/cp007_expression.tsv.gz'</span>, sep<span class="op">=</span><span class="st">'\t'</span>,
                         header<span class="op">=</span><span class="cn">T</span>, row.names<span class="op">=</span><span class="fl">1</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span>,
                         stringsAsFactors <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>

<span class="co"># Read in tree</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html" class="external-link">read.tree</a></span><span class="op">(</span><span class="st">'data/metastasis/cp007_tree.nwk.gz'</span><span class="op">)</span>
<span class="co"># Collapse one mutations with ape</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/collapse.singles.html" class="external-link">collapse.singles</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span>

<span class="co"># Read in meta data</span>
<span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">'data/metastasis/cp007_meta.tsv.gz'</span>, sep<span class="op">=</span><span class="st">'\t'</span>,
                   header<span class="op">=</span><span class="cn">T</span>, row.names<span class="op">=</span><span class="fl">1</span><span class="op">)</span>

<span class="co"># Signature file</span>
<span class="va">sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"h.all.v5.2.symbols.gmt"</span>, sep<span class="op">=</span><span class="st">"/"</span><span class="op">)</span></code></pre></div>
<p>A quick inspection of the meta data indicates that we have several
types of data associated with each of the leaves of this tree. Not only
do we have the anatomical site from which the cell was sampled
(<code>sampleID</code>) but also pre-computed statistics like the
Metastatic Rate (<code>scTreeMetRate</code>). In fact, users can specify
an arbitrary number of data items that have been precomputed. For
example, if a user has performed fitness inference as in <a href="https://www.biorxiv.org/content/10.1101/2021.10.12.464111v1" class="external-link">Yang
et al, bioRxiv 2021</a>, they can add these statistics as a column to
the meta data file.</p>
<p>In addition, a user might be interested in defining their own
clustering methodology on the tree. A simple clustering would entail
cutting the tree at a defined depth and grouping together cells that are
below each internal node at the defined depth (similar to the <a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/cutree" class="external-link"><code>cuttree</code></a>
method for grouping samples together from a hierarchical
clustering).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">clusterByDepth</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tree</span>, <span class="va">depth</span> <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">{</span>
  
  <span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>
  <span class="va">current_depth</span> <span class="op">=</span> <span class="fl">0</span>
  <span class="va">current_cluster_name</span> <span class="op">&lt;-</span> <span class="fl">1</span>
  <span class="va">root</span> <span class="op">&lt;-</span> <span class="fu">VISION</span><span class="fu">:::</span><span class="fu"><a href="../reference/find_root.html">find_root</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span>
  
  <span class="va">queue</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">root</span><span class="op">)</span>
  <span class="kw">while</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">queue</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span>
    
    <span class="kw">if</span> <span class="op">(</span><span class="va">current_depth</span> <span class="op">==</span> <span class="va">depth</span><span class="op">)</span> <span class="op">{</span>
      <span class="kw">for</span> <span class="op">(</span><span class="va">node</span> <span class="kw">in</span> <span class="va">queue</span><span class="op">)</span> <span class="op">{</span>
        <span class="va">clusters</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">current_cluster_name</span><span class="op">)</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span><span class="op">[</span><span class="fu">VISION</span><span class="fu">:::</span><span class="fu"><a href="../reference/get_all_children.html">get_all_children</a></span><span class="op">(</span><span class="va">tree</span>, <span class="va">node</span><span class="op">)</span><span class="op">]</span>
        <span class="va">current_cluster_name</span> <span class="op">&lt;-</span> <span class="va">current_cluster_name</span> <span class="op">+</span> <span class="fl">1</span>
      <span class="op">}</span>
      <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span>
    <span class="op">}</span>
    
    <span class="va">current_depth</span> <span class="op">&lt;-</span> <span class="va">current_depth</span> <span class="op">+</span> <span class="fl">1</span>
    
    <span class="va">new_queue</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span>
    <span class="kw">for</span> <span class="op">(</span><span class="va">node</span> <span class="kw">in</span> <span class="va">queue</span><span class="op">)</span> <span class="op">{</span>
      <span class="va">children</span> <span class="op">&lt;-</span> <span class="fu">VISION</span><span class="fu">:::</span><span class="fu"><a href="../reference/get_children.html">get_children</a></span><span class="op">(</span><span class="va">tree</span>, <span class="va">node</span><span class="op">)</span>
      <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">children</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span> <span class="op">&amp;&amp;</span> <span class="va">children</span> <span class="op">==</span> <span class="va">node</span><span class="op">)</span> <span class="op">{</span>
        <span class="va">clusters</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">current_cluster_name</span><span class="op">)</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span><span class="op">[[</span><span class="va">node</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
        <span class="va">current_cluster_name</span> <span class="op">&lt;-</span> <span class="va">current_cluster_name</span> <span class="op">+</span> <span class="fl">1</span>
      <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
        <span class="va">new_queue</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">new_queue</span>, <span class="va">children</span><span class="op">)</span>
      <span class="op">}</span>
    <span class="op">}</span>
    <span class="va">queue</span> <span class="op">&lt;-</span> <span class="va">new_queue</span>
  <span class="op">}</span>
  <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"Targeted depth is larger than the depth of the tree.
       Repeat call with smaller target depth."</span><span class="op">)</span>
<span class="op">}</span>

<span class="co"># add in depth based cluster annotations</span>
<span class="va">depth_clusters</span> <span class="op">&lt;-</span> <span class="fu">clusterByDepth</span><span class="op">(</span><span class="va">tree</span>, depth<span class="op">=</span><span class="fl">1</span><span class="op">)</span> 
<span class="va">meta</span><span class="op">$</span><span class="va">DepthClusters</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">cluster</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">depth_clusters</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">meta</span><span class="op">[</span><span class="va">depth_clusters</span><span class="op">[[</span><span class="va">cluster</span><span class="op">]</span><span class="op">]</span>, <span class="st">'DepthClusters'</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">cluster</span>
<span class="op">}</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Construct the Vision object</span>
<span class="va">projection_genes</span> <span class="op">&lt;-</span> <span class="fu">VISION</span><span class="fu">:::</span><span class="fu"><a href="../reference/filterGenesFano.html">filterGenesFano</a></span><span class="op">(</span><span class="va">expression</span><span class="op">)</span>
<span class="va">vis</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PhyloVision-class.html">PhyloVision</a></span><span class="op">(</span>tree<span class="op">=</span><span class="va">tree</span>, data<span class="op">=</span><span class="va">expression</span>, signatures<span class="op">=</span><span class="va">sig</span>,
                   meta<span class="op">=</span><span class="va">meta</span>, num_neighbors<span class="op">=</span><span class="fl">30</span>, projection_genes<span class="op">=</span> <span class="va">projection_genes</span><span class="op">)</span></code></pre></div>
<div class="section level4">
<h4 id="more-details-on-input">More details on input<a class="anchor" aria-label="anchor" href="#more-details-on-input"></a>
</h4>
<p><strong>Tree</strong></p>
<p>The <code>ape</code> phylogeny object. Singleton edges should be
collapsed prior to use. Ensure that leaf names correspond to the same
names used in the expression matrix.</p>
<p><strong>Expression Data</strong></p>
<p>The provided expression data should be library-normalized. The
expression data should not be log-transformed prior to loading into
VISION. For more information on the input to VISION, please refer to the
central <a href="VISION_vignette.html">VISION vignette</a>.</p>
<p><strong>Signatures</strong></p>
<p>Signatures can be provided as a list of paths to signature files
(*.gmt) or Signature objects.</p>
<p>See <a href="Signatures.html">the signature vignette</a> for more
information on finding or creating gene Signatures.</p>
<p><strong>Meta Data</strong></p>
<p>An R data.frame with cell-level meta-data. This could be confounding
variables (e.g. percentage of mitochondrial RNA, number of genes
detected) or experimental covariates (e.g. genotype, donor, batch).</p>
<p>This input is optional if Signatures are provided.</p>
<p><strong>Projection Genes</strong> These genes are used to infer
projections for clustering in the transcriptional space and 2D
visualizations.</p>
<p><strong>Other Options</strong></p>
<p>Other options and inputs can be provided to customize how VISION
runs.</p>
</div>
</div>
<div class="section level3">
<h3 id="running-phylovision-analysis">Running PhyloVision analysis<a class="anchor" aria-label="anchor" href="#running-phylovision-analysis"></a>
</h3>
<p>Next, we can perform the normal Vision analysis using the tree as the
latent space. In addition to the typical VISION pipeline, phyloAnlayze
will compute plasticity scores for each categorical meta data. For
example, in this case, a plasticity index on the <code>sampleID</code>
(i.e., which anatomical site a cell was harvested from) would correspond
to the scMetRate discussed in the original study.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">vis</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/phyloAnalyze-PhyloVision-method.html">phyloAnalyze</a></span><span class="op">(</span><span class="va">vis</span><span class="op">)</span></code></pre></div>
<p>Note: <code>phyloAnalyze</code> will add new entries to the
<code>object@metaData</code> slot that can be viewed on the interactive
report and evaluated with autocorrelation statistics. Specifically, a
default tree-based clustering will be added (named
<code>VISION_Cluster_Tree</code>) and numerical plasticity scores will
be computed as meta-data for each of the categorical data columns in the
user-specified <code>metaData</code>.</p>
</div>
<div class="section level3">
<h3 id="running-hotspot-analysis">Running Hotspot analysis<a class="anchor" aria-label="anchor" href="#running-hotspot-analysis"></a>
</h3>
<p>As mentioned above, PhyloVision can take advantage of the new Hotspot
functionality in VISION. Briefly, Hotspot is a tool for inferring
modules of genes that are significantly autocorrelated with one another
and a particular latent space (e.g., the first 30 principal components
of a gene expression matrix). When combined with PhyloVision, the
Hotspot functionality will use the user-defined phylogeny as the latent
space.</p>
<p>We can invoke the Hotspot analysis with the function
<code>runHotspot</code> and setting <code>tree=True</code>. Upon doing
so, this will identify modules of genes and add these as Signatures to
the VISION object for autocorrelation evaluation. Moreover, to add
interpretability to this analysis, VISION will also compute the
enrichment between all the user-defined Signatures and each Hotspot
module. This information will be accessible on the web-based report by
selecting the “Hotspot” mode in the top-right of the web-page.</p>
<p>The full Hotspot API is exposed for analysis as well. For more on the
Hotspot API see <a href="https://yoseflab.github.io/Hotspot/index.html" class="external-link">here</a>. To note,
while it is typically not recommended to use <code>model="normal"</code>
and <code>logdata=FALSE</code> in Hotpsot, we elect to do so because the
data here has been previously log-normalized. In more typical
single-cell anlayses where the user is working with a library-normalized
count matrix, we suggest setting <code>model="danb"</code> and
<code>logdata=TRUE</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">vis</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runHotspot.html">runHotspot</a></span><span class="op">(</span><span class="va">vis</span>, model<span class="op">=</span><span class="st">"danb"</span>, tree<span class="op">=</span><span class="cn">TRUE</span>, min_gene_threshold<span class="op">=</span><span class="fl">50</span>, n_neighbors <span class="op">=</span> <span class="fl">30</span>, number_top_genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">expression</span><span class="op">)</span>, logdata<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>We have additionally exposed the full Hotspot pipeline should a user
want to iterate with different parameters:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Init Hotspot</span>
<span class="va">hs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hsInit.html">hsInit</a></span><span class="op">(</span><span class="va">vis</span>, model <span class="op">=</span> <span class="st">"normal"</span>, tree<span class="op">=</span><span class="cn">TRUE</span>, logdata<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="co"># Init Hotspot KNN</span>
<span class="va">hs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hsCreateKnnGraph.html">hsCreateKnnGraph</a></span><span class="op">(</span><span class="va">hs</span>, <span class="va">vis</span>, n_neighbors<span class="op">=</span><span class="fl">30</span><span class="op">)</span>
<span class="co"># perform Hotspot analysis and store results in R</span>
<span class="va">hs_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hsComputeAutoCorrelations.html">hsComputeAutoCorrelations</a></span><span class="op">(</span><span class="va">hs</span>, number_top_genes<span class="op">=</span><span class="fl">12438</span><span class="op">)</span>
<span class="co"># Compute localcorr</span>
<span class="va">hs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hsComputeLocalCorrelations.html">hsComputeLocalCorrelations</a></span><span class="op">(</span><span class="va">hs</span>, <span class="va">hs_genes</span><span class="op">)</span>
<span class="co"># Calculate Hotspot Module Scores for informative genes</span>
<span class="va">hs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hsCalculateModuleScores.html">hsCalculateModuleScores</a></span><span class="op">(</span><span class="va">hs</span><span class="op">)</span>
<span class="co"># Cluster Hotspot modules and perform Vision based analysis on HS Modules and </span>
<span class="va">vis</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/analyzeHotspotObjectVision.html">analyzeHotspotObjectVision</a></span><span class="op">(</span><span class="va">vis</span>, <span class="va">hs</span>, tree<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>We can also access further Hotspot functionality using Reticulate and
Python.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/loadHotspotObject.html">loadHotspotObject</a></span><span class="op">(</span>bytes<span class="op">=</span><span class="va">vis</span><span class="op">@</span><span class="va">Hotspot</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/reticulate" class="external-link">reticulate</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/reticulate/man/use_python.html" class="external-link">use_python</a></span><span class="op">(</span><span class="st">'/usr/bin/python3'</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> hotspot</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>hs.plot_local_correlations()</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>plt.show()</span></code></pre></div>
<p>Note: the heatmap can also be visualized like so:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/loadHotspotObject.html">loadHotspotObject</a></span><span class="op">(</span>bytes<span class="op">=</span><span class="va">vis</span><span class="op">@</span><span class="va">Hotspot</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/EmilHvitfeldt/paletteer" class="external-link">paletteer</a></span><span class="op">)</span>
<span class="fu"><a href="../reference/draw_hotspot_heatmap.html">draw_hotspot_heatmap</a></span><span class="op">(</span><span class="va">hs</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="viewing-results">Viewing Results<a class="anchor" aria-label="anchor" href="#viewing-results"></a>
</h3>
<p>Finally, we can launch the Vision browser. If you are local, the
following will work:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/viewResults.html">viewResults</a></span><span class="op">(</span><span class="va">vis</span><span class="op">)</span></code></pre></div>
<p>Else, if you would like to launch from a server, you can use the
following settings:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/viewResults.html">viewResults</a></span><span class="op">(</span><span class="va">vis</span>, host<span class="op">=</span><span class="st">'0.0.0.0'</span>, port<span class="op">=</span><span class="fl">8100</span>, browser<span class="op">=</span><span class="cn">F</span><span class="op">)</span></code></pre></div>
<p>Now, the report will be visible from port <code>8100</code> from your
server (e.g., <a href="http://server-name:8100/Results.html" class="external-link uri">http://server-name:8100/Results.html</a>)</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Matt Jones, David Detomaso, Tal Ashuach, Yanay
Rosen.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
