<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to VISION • VISION</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to VISION">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">VISION</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.99.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/VISION-vignette.html">Introduction to VISION</a>
    </li>
    <li>
      <a href="../articles/micropooling.html">Micropooling</a>
    </li>
    <li>
      <a href="../articles/trajectories.html">Incorporating a Trajectory Model</a>
    </li>
    <li>
      <a href="../articles/web_only/SignatureConsistency.html">Signature Consistency</a>
    </li>
    <li>
      <a href="../articles/web_only/Signatures.html">Gene Signatures</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/yoseflab/VISION">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Introduction to VISION</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/yoseflab/VISION/blob/master/vignettes/VISION-vignette.Rmd"><code>vignettes/VISION-vignette.Rmd</code></a></small>
      <div class="hidden name"><code>VISION-vignette.Rmd</code></div>

    </div>

    
    
<div id="preliminaries" class="section level1">
<h1 class="hasAnchor">
<a href="#preliminaries" class="anchor"></a>Preliminaries</h1>
<p>If you have yet to install VISION, we recommend installing the package from github to install this pacakge. Full source code can be found at the VISION github repository, available <a href="http://www.github.com/YosefLab/VISION">here</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">require</span>(devtools)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="st">"YosefLab/VISION"</span>)</a></code></pre></div>
<p>Once VISION and R are installed, you may load in VISION using <code>library(VISION)</code>.</p>
</div>
<div id="using-vision" class="section level1">
<h1 class="hasAnchor">
<a href="#using-vision" class="anchor"></a>Using VISION</h1>
<p>Running an analysis with vision consists of three steps:</p>
<ol style="list-style-type: decimal">
<li>Creating the VISION object</li>
<li>Running the <code>analyze</code> function</li>
<li>Browsing results</li>
</ol>
<div id="creating-the-vision-object" class="section level2">
<h2 class="hasAnchor">
<a href="#creating-the-vision-object" class="anchor"></a>Creating the VISION object</h2>
<p>Creating the VISION object requires a gene expression matrix and <em>either</em> a list of Gene Signatures or a data.frame of meta-data.</p>
<p>In this example - both are provided.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># Load VISION</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">library</span>(VISION)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co"># Read in expression counts (Genes X Cells)</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5">counts &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">"data/expression_counts.txt.gz"</span>,</a>
<a class="sourceLine" id="cb2-6" data-line-number="6">                     <span class="dt">header =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb2-7" data-line-number="7">                     <span class="dt">sep =</span> <span class="st">'</span><span class="ch">\t</span><span class="st">'</span>,</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">                     <span class="dt">row.names =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb2-9" data-line-number="9"></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="co"># Scale counts within a sample</span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11">n.umi &lt;-<span class="st"> </span><span class="kw">colSums</span>(counts)</a>
<a class="sourceLine" id="cb2-12" data-line-number="12"></a>
<a class="sourceLine" id="cb2-13" data-line-number="13">scaled_counts &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">t</span>(counts) <span class="op">/</span><span class="st"> </span>n.umi) <span class="op">*</span><span class="st"> </span><span class="kw">median</span>(n.umi)</a>
<a class="sourceLine" id="cb2-14" data-line-number="14"></a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="co"># Read in meta data (Cells x Vars)</span></a>
<a class="sourceLine" id="cb2-16" data-line-number="16">meta =<span class="st"> </span><span class="kw">read.table</span>(<span class="st">"data/glio_meta.txt.gz"</span>, <span class="dt">sep=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, <span class="dt">header=</span>T, <span class="dt">row.names=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb2-17" data-line-number="17"></a>
<a class="sourceLine" id="cb2-18" data-line-number="18">vis &lt;-<span class="st"> </span><span class="kw"><a href="../reference/VISION-class.html">Vision</a></span>(scaled_counts, </a>
<a class="sourceLine" id="cb2-19" data-line-number="19">              <span class="dt">signatures =</span> <span class="kw">c</span>(<span class="st">"data/h.all.v5.2.symbols.gmt"</span>), </a>
<a class="sourceLine" id="cb2-20" data-line-number="20">              <span class="dt">meta =</span> meta)</a></code></pre></div>
<p><strong>Expression Data</strong></p>
<p>The provided expression data should be scaled and normalized. The example above shows just a simple UMI-scaling, but it is recommended to apply more advanced normalization procedures such as batch correction or removal of technical confounders.</p>
<p>The expression data should not be log-transformed prior to loading into VISION.</p>
<p><strong>Signatures</strong></p>
<p>Signatures can be provided as a list of paths to signature files (*.gmt) or Signature objects.</p>
<p>See <a href="Signatures.html">the signature vignette</a> for more information on finding or creating gene Signatures.</p>
<p><strong>Meta Data</strong></p>
<p>An R data.frame with cell-level meta-data. This could be confounding variables (e.g. percentage of mitochondrial RNA, number of genes detected) or experimental covariates (e.g. genotype, donor, batch).</p>
<p>This input is optional if Signatures are provided.</p>
<p><strong>Other Options</strong></p>
<p>Other options and inputs can be provided to customize how VISION runs. For information on this, see the “Customizing VISION Analysis” section below.</p>
</div>
<div id="running-an-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#running-an-analysis" class="anchor"></a>Running an Analysis</h2>
<p>To run an analysis, simply call the analyze function:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># Set the number of threads when running parallel computations</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw">options</span>(<span class="dt">mc.cores =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"></a>
<a class="sourceLine" id="cb3-4" data-line-number="4">vis &lt;-<span class="st"> </span><span class="kw"><a href="../reference/analyze-Vision-method.html">analyze</a></span>(vis)</a></code></pre></div>
</div>
<div id="viewing-results" class="section level2">
<h2 class="hasAnchor">
<a href="#viewing-results" class="anchor"></a>Viewing Results</h2>
<p>With the processed Vision object, a dynamic web report can be generated with the <code>viewResults</code> function.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="../reference/viewResults.html">viewResults</a></span>(vis)</a></code></pre></div>
<p>This will launch a browser running the interactive report.</p>
<p>Other options (port, host, browser) can be provided to control how this occurs. For example, if you are launching a report on a remote server (such as an AWS instance) and want to make it accessible to others, run this with <code>host="0.0.0.0"</code>, some selected port number (e.g. <code>port=8888</code>), and <code>browser=FALSE</code> (so a browser isn’t auto-opened). Then the report should be available at “&lt;your instance IP address&gt;:8888”. (Note: You will also likely need to enable inbound traffic on your selected port for this to work correctly).</p>
<p>Alternately, you can work with the VISION object directly in R. For example:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># Display autocorrelation coefficients for signatures</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw">head</span>(vis<span class="op">@</span>SigConsistencyScores<span class="op">@</span>sigProjMatrix)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co"># View autocorrelation empirical p-values</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="kw">head</span>(vis<span class="op">@</span>SigConsistencyScores<span class="op">@</span>emp_pMatrix)</a>
<a class="sourceLine" id="cb5-6" data-line-number="6"></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co"># Plot signature scores for a signature of interest</span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8">tsne &lt;-<span class="st"> </span>vis<span class="op">@</span>Projections<span class="op">$</span>tSNE30</a>
<a class="sourceLine" id="cb5-9" data-line-number="9">sigScores &lt;-<span class="st"> </span>vis<span class="op">@</span>sigScores[, <span class="st">"HALLMARK_INTERFERON_GAMMA_RESPONSE"</span>]</a>
<a class="sourceLine" id="cb5-10" data-line-number="10"></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>() <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">x=</span>tsne[, <span class="dv">1</span>], <span class="dt">y=</span>tsne[, <span class="dv">2</span>], <span class="dt">color=</span>sigScores) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>()</a></code></pre></div>
<p>For more details on the structure of the VISION object see XXX.</p>
</div>
<div id="customizing-the-latent-space" class="section level2">
<h2 class="hasAnchor">
<a href="#customizing-the-latent-space" class="anchor"></a>Customizing the Latent Space</h2>
<p>VISION requires a latent space to model the similarity between cells (used to determine a cell’s local neighborhood).</p>
<p>By default, this is calculated via PCA on a subset of the genes. A few arguments control this process:</p>
<ul>
<li>
<code>projection_genes</code> - controls how genes are selected for PCA. Options are:
<ul>
<li>
<code>"threshold"</code> - use a threshold to select genes. The <code>threshold</code> argument specifies either the number or proportion of genes in which a cell must be expressed to be included.</li>
<li>
<code>"fano"</code> - first the “threshold” filter is applied. Then genes are ordered by mean expression into 30 bins and within each bin, genes with a high fano factor (2 MAD above the median) are retained</li>
<li>character vector of gene names - specify the genes to use directly</li>
</ul>
</li>
</ul>
<div id="specifying-a-custom-latent-space" class="section level3">
<h3 class="hasAnchor">
<a href="#specifying-a-custom-latent-space" class="anchor"></a>Specifying a Custom Latent Space</h3>
<p>Alternately, if a latent space has been computed elsewhere (either via PCA, or other factor analysis methods such as <a href="https://github.com/epierson9/ZIFA">ZIFA</a>, <a href="https://github.com/drisso/zinbwave">ZINB-WaVE</a> or <a href="https://github.com/YosefLab/scVI">scVI</a>) it can be provided via the <code>latentSpace</code> argument as a data frame.</p>
</div>
</div>
<div id="micropooling" class="section level2">
<h2 class="hasAnchor">
<a href="#micropooling" class="anchor"></a>Micropooling</h2>
<p>scRNA-seq experiments have grown in size over the past couple of years, and as such we have provided an algorithm for pooling together similar cells and continuing with analysis, thus reducing the cell-wise time complexity of the VISION pipeline. For more information, please see our micropooling vignette.</p>
<p>You can control the parameters of the micropooling algorithm using two arguments:</p>
<ul>
<li>
<code>pool</code> - boolean specifying whether or not to apply micropooling. By default this is set to ‘auto’ and micropooling is run when the number of cells exceeds 15,000.</li>
<li>
<code>cellsPerPartition</code> - integer specifying the maximum number of cells per micropool.</li>
</ul>
</div>
<div id="adding-2d-projections" class="section level2">
<h2 class="hasAnchor">
<a href="#adding-2d-projections" class="anchor"></a>Adding 2d Projections</h2>
<p>Two-dimensional projections are used to visualize the data in the output report.</p>
<p>By default, VISION computes tSNE on the latent space for visualization. However, other options are availabled via the <code>projection_methods</code> argument.</p>
<p>Often times, users will have pre-computed projections that they would like to use for visualizing their data (e.g. a pre-computed tSNE, or UMAP projetction). In this case, the <code>addProjection</code> method can be used to add this view of the data to the output report.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">projection &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">"umap_results.csv"</span>)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co"># projection is a matirx or data.frame of dimension (Cells x 2)</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"></a>
<a class="sourceLine" id="cb6-5" data-line-number="5">vis &lt;-<span class="st"> </span><span class="kw"><a href="../reference/addProjection-Vision-method.html">addProjection</a></span>(vis, <span class="st">"UMAP"</span>, projection)</a></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#preliminaries">Preliminaries</a></li>
      <li>
<a href="#using-vision">Using VISION</a><ul class="nav nav-pills nav-stacked">
<li><a href="#creating-the-vision-object">Creating the VISION object</a></li>
      <li><a href="#running-an-analysis">Running an Analysis</a></li>
      <li><a href="#viewing-results">Viewing Results</a></li>
      <li><a href="#customizing-the-latent-space">Customizing the Latent Space</a></li>
      <li><a href="#micropooling">Micropooling</a></li>
      <li><a href="#adding-2d-projections">Adding 2d Projections</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Matt Jones, David Detomaso, Tal Ashuach.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
